charts/fluent-bit/templates/_pod.tpl

```diff
@@ -37,9 +37,19 @@ containers:
   {{- end }}
     image: "{{ .Values.image.repository }}:{{ default .Chart.AppVersion .Values.image.tag }}"
     imagePullPolicy: {{ .Values.image.pullPolicy }}
-  {{- if .Values.env }}
+  {{- if or .Values.env .Values.envWithTpl }}
     env:
+      {{- if .Values.env }}
       {{- toYaml .Values.env | nindent 6 }}
+      {{- end }}
+      {{- if .Values.envWithTpl }}
+      {{- with .Values.envWithTpl }}
+      {{- range $key, $val := . }}
```

Comment:
```suggestion
      {{- range $key, $val := . }}
```
You don't need the `if` and `with`.

Reply:
Indeed, those were not needed, but when I removed them I was unable to get the `range` function working, so I've switched to be simply using `{{- tpl (toYaml .) $ | nindent 6 }}`